### PR TITLE
Add rgb-light and encoder for lily58 mod

### PR DESCRIPTION
**Pull request by LVladymyr**
Add rgb-light and encoders for [lily58-glow-enc mod](https://github.com/orvisevans/Lily58-Glow-Enc)

## Description

Set custom pins and settings for enabling rgb-light and encoders (one per controller).

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/lily58/glowEnc/glowEnc.h [hunk 1]
```diff
@@ -0,0 +1,50 @@
+#pragma once
```
Comment (drashna) [inline keyboards/lily58/glowEnc/glowEnc.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/lily58/glowEnc/config.h [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+#pragma once
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFC51
+#define PRODUCT_ID      0x0059
+#define DEVICE_VER      0x0100
+#define MANUFACTURER    ORVISEVANS+
+#define PRODUCT         Lily58GlowEnc
+#define DESCRIPTION     Lily58 is 6×4+5keys column-staggered split keyboard with glowing keys and encoders.
```
Comment (drashna) [inline keyboards/lily58/glowEnc/config.h:29]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]
please rename the folder and files to be all lower case.

**Review by drashna — APPROVED** [review]

**Review by noroadsleft — APPROVED** [review]
Looks good to me.

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**